internal/provider/group_resource.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int32planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	netbird "github.com/netbirdio/netbird/management/client/rest"
+	"github.com/netbirdio/netbird/management/server/http/api"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &Group{}
+var _ resource.ResourceWithImportState = &Group{}
+
+func NewGroup() resource.Resource {
+	return &Group{}
+}
+
+// Group defines the resource implementation.
+type Group struct {
+	client *netbird.Client
+}
+
+// GroupModel describes the resource data model.
+type GroupModel struct {
+	Id             types.String `tfsdk:"id"`
+	Name           types.String `tfsdk:"name"`
+	Peers          types.List   `tfsdk:"peers"`
+	Resources      types.List   `tfsdk:"resources"`
+	PeersCount     types.Int32  `tfsdk:"peers_count"`
+	ResourcesCount types.Int32  `tfsdk:"resources_count"`
+	Issued         types.String `tfsdk:"issued"`
+}
+
+func (r *Group) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+func (r *Group) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Create and assign Groups, see [NetBird Docs](https://docs.netbird.io/how-to/manage-network-access#groups) for more information.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Group ID",
+				Computed:            true,
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Group name identifier",
+				Required:            true,
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"peers_count": schema.Int32Attribute{
```

Comment:
it might be useful to not put those 2 `*_count` fields into the provider, because we can just do  `length(*.peers)`. I'd rather see only one way to do this

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7